app/service/Social.py

```diff
@@ -36,18 +41,43 @@ async def create_post(self, input_post: PostCreateSchema) -> PostSchema:
         user: ReducedUser = ReducedUser.from_pydantic(get_user)
         post = Post.from_pydantic(input_post)
         id_post = self.social_repository.add_post(post)
-        crated_post = self.social_repository.get_post(id_post)
-        map_author_user_id(user, crated_post)
-        return PostSchema.model_validate(crated_post)
+        created_post = self.social_repository.get_post(id_post)
+        print("created: ", created_post)
+        map_author_user_id(user, created_post)
+        print("created w user: ", created_post)
+        return PostSchema.model_validate(created_post)
 
     async def get_post(self, id_post: int) -> PostSchema:
         post: Post = self.social_repository.get_post(id_post)
         if post is None:
             raise ItemNotFound("Post", id_post)
+
         get_user: GetUserSchema = await UserService.get_user(post["author_user_id"])
         user: ReducedUser = ReducedUser.from_pydantic(get_user)
         map_author_user_id(user, post)
-        return PostSchema.model_validate(post)
+
+        final_comments = []
+
+        for comment in post['comments']:
```

Comment:
Ojo con este for porque estas haciendo N fetchs a User, lo cual puede escalar muy mal si una publicacion tiene muchisimos comentarios.

En [esta funcion de User Service](https://github.com/Hanagotchi/social/blob/0efbbb6b3c35f2f8929bd999a9287af0d1599066/app/service/Users.py#L49) tenes un endpoint que le pasas la lista de ids y te devuelve todos los users de una. Es mas prolijo usar esto.

Reply:
si uso eso tengo que recorrer primero todos los comentarios para sacar los ids, despues pegarle al  UserService.get_user y despues volver a recorrer los comentarios para mappear el user con su comentario (y eso asumiendo que los devuelve en el mismo orden en el que tengo la lista de comentarios)
me parece mas desprolijo, agrega complejidad y no considero que vayamos a tener problemas de performance